component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java

```diff
@@ -1903,6 +1932,46 @@ public String name() {
             }
         }
 
+        @Override
+        public void onDriverRunner(final Class<?> type, final DriverRunner processor) {
+            final Constructor<?> constructor = findConstructor(type);
+            final Supplier<List<ParameterMeta>> parameterMetas = lazy(() -> executeInContainer(plugin,
+                    () -> parameterModelService
+                            .buildParameterMetas(constructor, getPackage(type),
+                                    new BaseParameterEnricher.Context(LocalConfiguration.class
+                                            .cast(services.getServices().get(LocalConfiguration.class))))));
+            final Function<Map<String, String>, Object[]> parameterFactory =
+                    createParametersFactory(plugin, constructor, services.getServices(), parameterMetas);
+            final String name = of(processor.name()).filter(n -> !n.isEmpty()).orElseGet(type::getName);
+            final ComponentFamilyMeta component = getOrCreateComponent(processor.family());
+            final Function<Map<String, String>, org.talend.sdk.component.runtime.standalone.DriverRunner> instantiator =
+                    context.getOwningExtension() != null && context
+                            .getOwningExtension()
+                            .supports(org.talend.sdk.component.runtime.standalone.DriverRunner.class)
+                                    ? config -> executeInContainer(plugin,
+                                            () -> context
+                                                    .getOwningExtension()
+                                                    .convert(new ComponentInstanceImpl(doInvoke(
+                                                            constructor, parameterFactory.apply(config)), plugin,
+                                                            component.getName(), name),
+                                                            org.talend.sdk.component.runtime.standalone.DriverRunner.class))
+                                    : config -> new DriverRunnerImpl(this.component.getName(), name, plugin,
+                                            doInvoke(constructor, parameterFactory.apply(config)));
```

Comment:
Would be easier to understand with code like : 
```java
final boolean isDriverSupport = context.getOwningExtension() != null && context
                            .getOwningExtension()
                            .supports(org.talend.sdk.component.runtime.standalone.DriverRunner.class);

final Function<Map<String, String>, org.talend.sdk.component.runtime.standalone.DriverRunner> instantiator =
    (Map<String, String> config) -> {
            final Serializable instance = doInvoke(constructor, parameterFactory.apply(config));
            if (isDriverSupport) {
                     final ComponentInstanceImpl cpnInstance = new ComponentInstanceImpl(instance, plugin,
                                                            component.getName(), name);
                     return executeInContainer(plugin,
                               context.getOwningExtension()
                                           .convert(cpnInstance,
                                                            org.talend.sdk.component.runtime.standalone.DriverRunner.class))
              }
              else {
                       new DriverRunnerImpl(this.component.getName(), name, plugin,   instance);
            }
   }
```
but that's my poor brain which is not adapt enough :)